core/templates/domain/objects/NumberWithUnitsObjectFactory.ts

```diff
@@ -124,6 +124,15 @@ export class NumberWithUnits {
       units: this.units
     };
   }
+
+  canonicalizeUnits(): Unit[] {
```

Comment:
getCanonicalRepresentationOfUnits()

Also, you should have a bunch of unit tests for this.

Reply:
Done.